teps/0048-task-results-without-results.md

```diff
@@ -426,47 +429,77 @@ will fix the `Pipeline`.
 
 ## Proposal
 
-We propose adding an optional field - `default` - to the `Result` specification in `Task`.
+We propose adding default value at the place where variable replacement
+happens and let consumer decide which value it wants to pick up. For example:
 
 ```yaml
 apiVersion: tekton.dev/v1beta1
 kind: Task
 metadata:
   name: task
 spec:
-  results:
-  - name: merge_status
-    description: whether to rebase or squash
-    type: string
-    default: rebase
-  - name: branches
-    description: branches in the repository
-    type: array
-    default:
-    - main
-    - v1alpha1
-    - v1beta1
-    - v1
-  - name: images
-    type: object
-    properties:
-      node:
-       type: string
-       default: "node:latest" 
-      gcloud:
-       type: string
-       default: "gcloud:latest"
   steps:
   ...
+  results:
+    - name: merge_status
+      description: whether to rebase or squash
+      type: string
+    - name: branches
+      description: branches in the repository
+      type: array
+    - name: images
+      type: object
+      properties:
+        node:
+          type: string
+        gcloud:
+          type: string
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: pipeline-with-defaults
+spec:
+  tasks:
+    - name: taskA
+      onError: continue
+      taskRef:
+        name: task
+      results:
+      - name: merge_status
+        default: squash
+    - name: taskB
+      runAfter:
+      - "taskA"
+      taskRef:
+        name: demo-task
+      params:
+      - name: param1
+        value: $(tasks.taskA.results.merge_status || "foobar")
+      - name: param2
+        value: $(tasks.taskA.results.branches || [foo, bar, baz])
+      - name: param3
+        value: $(tasks.taskA.results.images || {node="foo", gcloud="bar"})
+    - name: taskC
+      runAfter:
+      - "taskA"
+      taskRef:
+        name: demo-task
+      params:
+      - name: param1
+        value: $(tasks.taskA.results.merge_status)
+      - name: param2
+        value: $(tasks.taskA.results.branches || [foo, bar, baz])
+      - name: param3
+        value: $(tasks.taskA.results.images || {node="foo", gcloud="bar"})
+  ...
 ```
 
-
-Adding a default value to `Results` is optional; validation of a `Task` doesn't
-fail if the default value isn't provided.
-
-Adding a default value to a `Result` will guarantee that it will hold a value even
-if the `Task` fails to produce it. If a `Task` does not produce a `Result` that does not
-have a default value, then the `Task` should fail - see [tektoncd/pipeline#3497](https://github.com/tektoncd/pipeline/issues/3497) for further details.
+If the `Task` doesn't produces any result then it should fail. In order to
+continue with the execution of `Pipeline`, `onError: continue` should be added
```

Comment:
I thought `onError` was only available in a task spec? can you give an example?

Reply:
not sure whether [TEP-0050](https://github.com/tektoncd/community/blob/main/teps/0050-ignore-task-failures.md#proposal) is 100% implemented or not but saw it there cc @jerop

Reply:
It's still "implementable" instead of "implemented" so maybe it wasn't completed? cc @QuanZhang-William

Reply:
Yeah, this is correct. I didn't get a chance to implement this feature. I also did a PR search in the pipeline and didn't find much related.